### PR TITLE
Expose profile and follow

### PR DESCRIPTION
**Pull request by larribas**
I've exposed the remaining user and profile endpoints and made some opinionated refactors, such as:

- Commands and queries are stored and registered in the same way for all cells, including the API cell. Since there will be a big index of c/q that are required and registered, I believe it is better to keep it isolated but together (it favors quick find, delete, add, and rename ops).
- Router and event broker implementation don't need extra tech-dependent considerations such as attributes or channels, and the names they use are not tied to any technology.
- I've reverted to the `bio` and `image` attributes being non-nullable. I didn't realize when I merged your PR, but there is a dichotomy we need to address gracefully. This is what I wrote for the commit description:

> nil represents "i don't care about this attribute", whereas the empty string
means "i explicitly want this optional attribute to be empty". I think this
makes sense because it follows a commonplace distinction between zero values and
nil pointers that comes from low-level languages like C or Go.

If we were to use `nil` to represent optional attributes in the domain we would need a custom type `Unspecified` in the HTTP pipeline.

Let me know what you think ;)

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -3,8 +3,6 @@ services:
   api:
     environment:
       - API_AUTH_TOKEN_SECRET=dont_tell_anyone
-    ports:
-      - "3000:3000"
```
Comment (hecrj) [inline docker-compose.dev.yml:7]: I sometimes used `./bin/start` after `./bin/dev` to quickly manually test. Now that the pipeline is set up we probably do not need it anymore.

File: lib/real_world/api/http/mappers/route.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+              else
+                [nil, error(:param_not_found, name: name)]
+              end
+            end
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (hecrj) [inline lib/real_world/api/http/mappers/route.rb:27]: Nice, you got the idea!

What do you think about this? I think it's pretty simple and clean!

File: lib/real_world/api/commands/api_update_current_user.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+          capture_errors(updated_user) do
+            user = updated_user.fetch(:data)
+            response(user: user.merge(token: auth_token))
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (hecrj) [inline lib/real_world/api/commands/api_update_current_user.rb:29]: After working with Elm so much time, I am really not a fan of having so many files. Is there any reasoning behind it?

File: lib/real_world/api/queries/api_get_profile.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  module Api
+    module Queries
+      class ApiGetProfile < Colmena::Command
+        def call(auth_token:, username:)
+          token, error = port(:tokens).decode_auth(auth_token)
+          return error_response(:forbidden, reason: error) if error
+
+          read_user = port(:router).query(:read_user_by_username).call(
+            username: username,
+          )
```
Comment (hecrj) [inline lib/real_world/api/queries/api_get_profile.rb:15]: Missing a blank line here?

File: lib/real_world/ports/router/spec_shared_examples.rb [hunk 5]
```diff
@@ -27,7 +27,7 @@ def self.name
 
       def call(id:, message:)
         port(:repository).set(id, message)
-        response(true, events: [event(:message_written))
+        response(true, events: [event(:message_written)])
```
Comment (hecrj) [inline lib/real_world/ports/router/spec_shared_examples.rb:30]: How did this work? :O

**Review by hecrj — APPROVED** [review]

File: lib/real_world/api/http/mappers/route.rb (hunk 2 above)
Comment (larribas) [reply lib/real_world/api/http/mappers/route.rb:27]: Yeah, I like it a lot

File: lib/real_world/api/commands/api_update_current_user.rb (hunk 3 above)
Comment (larribas) [reply lib/real_world/api/commands/api_update_current_user.rb:29]: I think the filesystem makes for a pretty natural command/query/listener index. This is what I've experienced with our particular ruby codebase:

* I want to jump to a query. I usually know the name of the query. If all queries are in the same file, I need to start thinking in terms of which cell they're in, and whether it's a query or a command, so mentally is takes me a bit longer.
* This way I immediately know which cells are tested and can go to that spec.
* This way I can remove a command almost without any hassle.
* This way I can add a new command without worrying about the order of declarations within the same file, and so on.

It's probably related to the way I move with vim/emacs too, but I think it's a nicer structure for Colmena.

When I think about doing the same thing in Elm/Haskell, it seems quite overkill, but then again I would need to make sure my [ctags](https://en.wikipedia.org/wiki/Ctags) are set up properly if I want to remain this productive.

File: lib/real_world/ports/router/spec_shared_examples.rb (hunk 5 above)
Comment (larribas) [reply lib/real_world/ports/router/spec_shared_examples.rb:30]: I must have forgotten to rerun the tests after changing the name of the event or something like that.

**Pull request merged**